src/nvmet.c

```diff
@@ -802,58 +1153,133 @@ int nvmet_enable_port(unsigned char *interface, int port, int protocol)
 
 	return SUCCESS;
 }
 
-int nvmet_disable_port(int port)
+/**
+ * This function removes the specified NVMe Target port
+ *
+ * @param port The port number to be disabled.
+ * @param return_message This is a pointer to a buffer that will be filled with a message
+ *
+ * @return The return code of the function.
+ */
+int nvmet_disable_port(int port, char *return_message)
 {
 	int rc = INVALID_VALUE, n, err;
-	unsigned char path[NAMELEN] = {0x0};
+	char path[NAMELEN] = {0x0};
 	struct dirent **list;
+	char *msg;
 
 	sprintf(path, "%s/%d", SYS_NVMET_PORTS, port);
 	if (access(path, F_OK) != SUCCESS) {
-		printf("Error. NVMe Target Port %d does not exist.\n", port);
+		msg = "Error. NVMe Target Port %d does not exist";
+		print_error(msg, return_message, port);
 		return rc;
 	}
 
 	/* Make sure that no subsystems are mapped from this port. */
-        sprintf(path, "%s/%d/subsystems/", SYS_NVMET_PORTS, port);
-        if ((err = scandir(path, &list, NULL, NULL)) < 0) {
-                printf("Error. Unable to access %s.\n%s: scandir: %s\n", path, __func__, strerror(errno));
-                return rc;
-        }
-        for (n = 0; n < err; n++) if (list[n] != NULL) free(list[n]);    /* clear list */
-        if (err > 2) {
-                printf("This port is currently in use.\n");
-                return rc;
-        }
+	sprintf(path, "%s/%d/subsystems/", SYS_NVMET_PORTS, port);
+	if ((err = scandir(path, &list, NULL, NULL)) < 0) {
+		msg = "Error. Unable to access %s. %s: scandir: %s";
+		print_error(msg, return_message, path, __func__, strerror(errno));
+		return rc;
+	}
+	list = clean_scandir(list, err);
+	if (err > 2) {
+		msg = "%s";
+		print_error(msg, return_message, "This port is currently in use.");
+		return rc;
+	}
 
 	/* Remove the port */
 	sprintf(path, "%s/%d", SYS_NVMET_PORTS, port);
 	if ((rc = rmdir(path)) != SUCCESS) {
-		printf("Error. Unable to remove port.\n%s: rmdir: %s\n", __func__, strerror(errno));
+		msg = "Error. Unable to remove port. %s: rmdir: %s";
+		print_error(msg, return_message, __func__, strerror(errno));
 		return rc;
 	}
-
-	printf("NVMe Target port %d has been removed.\n", port);
+	msg = "NVMe Target port %d has been removed.";
+	print_error(msg, return_message, port);
 
 	return SUCCESS;
 }
+
+#else
+
+/**
+ * This function scans the system for NVMe-oF subsystems and ports, and then creates a JSON string that contains the
+ * information
+ *
+ * @param error_message This is a pointer to a character array that will contain the error message if the function fails.
+ * @param json_result This is the JSON string that will be returned to the caller.
+ */
+int nvmet_view_exports_json(char *error_message, char **json_result) {
+	int rc = SUCCESS;
+	struct NVMET_PROFILE *nvmet = NULL;
+	struct NVMET_PORTS *ports = NULL;
+
+	nvmet = nvmet_scan_subsystem(error_message);
+	if ((nvmet == NULL) && (strlen(error_message) != 0)) {
+//		print_message(INVALID_VALUE, error_message, json_flag);
```

Comment:
Is this something we should enable for enabled verbosity or should we remove commented code?

Reply:
Removed commented code.